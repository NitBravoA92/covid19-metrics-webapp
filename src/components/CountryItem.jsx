import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { BsArrowRightCircle } from 'react-icons/bs';

const CountryItem = ({
  image, name, cases,
}) => {
  const countryName = name.replace(/\s/g, '-');

  return (
    <NavLink to={`${countryName}`} className="country-item child-element">
      <div className="image">
        <img src={image} alt={name} loading="lazy" className="country-flag" />
      </div>
      <div className="country-info">
        <h3 className="name">{name}</h3>
        <p className="cases-stats">
          {cases}
          {' '}
          cases
        </p>
      </div>
      <div className="footer-icon">
        <BsArrowRightCircle />
      </div>
    </NavLink>
  );
};

CountryItem.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  cases: PropTypes.string.isRequired,
};

export default CountryItem;
