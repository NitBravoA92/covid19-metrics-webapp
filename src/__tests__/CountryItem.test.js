import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, screen } from '@testing-library/react';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import CountryItem from '../components/CountryItem';

describe('The CountryItem component', () => {
  test('should render correctly into the DOM', () => {
    const { CountryItemComponent } = render(
      <MemoryRouter>
        <CountryItem image="https://disease.sh/assets/img/flags/af.png" name="Afghanistan" cases="224,317" />
      </MemoryRouter>,
    );
    expect(CountryItemComponent).toMatchSnapshot();
  });

  test('should navigate to the third page when user clicked on Albania country item', () => {
    render(
      <BrowserRouter>
        <CountryItem image="https://disease.sh/assets/img/flags/al.png" name="Albania" cases="334,726" />
      </BrowserRouter>,
    );
    const CountryItemLink = document.querySelector('.country-item');
    fireEvent.click(CountryItemLink);
    expect(window.location.pathname).toBe('/Albania');
  });

  test('should render the text: Algeria and the number of cases', () => {
    render(
      <MemoryRouter>
        <CountryItem image="https://disease.sh/assets/img/flags/dz.png" name="Algeria" cases="271,852" />
      </MemoryRouter>,
    );
    const CountryItemTitle = screen.getByText(/Algeria/i);
    const CountryItemCases = screen.getByText(/271,852 cases/i);

    expect(CountryItemTitle).toBeInTheDocument();
    expect(CountryItemCases).toBeInTheDocument();
  });
});
