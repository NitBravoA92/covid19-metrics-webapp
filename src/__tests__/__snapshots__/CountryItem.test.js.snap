// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CountryItem component should render correctly into the DOM 1`] = `undefined`;
